src/Models/Command/CommandResponse.cs

```diff
@@ -20,8 +22,45 @@ public class CommandResponse
 
     [JsonPropertyName("results")]
     [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-    public object? Results { get; set; }
+    public ResponseResult? Results { get; set; }
 
     [JsonPropertyName("duration")]
     public long Duration { get; set; }
+}
+
+[JsonConverter(typeof(ResultConverter))]
+public sealed class ResponseResult
```

Comment:
Consider moving this into a separate file?

Reply:
Should we change the invocation style to hide the use of the local serialization class?

```cs
context.Response.SetResult(result, JsonSourceGenerationContext.Default.KeyValueDeleteCommandResult);
```

Reply:
```cs
if (items?.Count > 0)
{
    context.Response.SetResult(new ItemQueryCommandResult(items), JsonSourceGenerationContext.Default.ItemQueryCommandResult);
} else {
    context.Response.SetResult(null);
}
```

Reply:
I could do that, but I usually keep types that are serialized as simple as possible. If I add a SetResult API I assume I would need to make the Response private, so it wouldn't be accessible. But then it wouldn't be serialized automatically, right?